File: test_cobalt.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

void main() async {
  final uri = Uri.parse('https://api.cobalt.tools/api/json');
  final res = await http.post(
    uri,
    headers: {
      'Accept': 'application/json',
      'Content-Type': 'application/json',
    },
    body: json.encode({
      'url': 'https://www.youtube.com/watch?v=2Vv-BfVoq4g',
      'vCodec': 'h264',
      'vQuality': '720',
      'isAudioOnly': false,
    }),
  );
  print('Cobalt Status: ${res.statusCode}');
  print('Cobalt Body: ${res.body}');
}

File: test_cobalt.py
import urllib.request
import json

def test_api():
    try:
        req = urllib.request.Request(
            'https://co.wuk.sh/api/json',
            data=json.dumps({
                "url": "https://www.youtube.com/watch?v=2Vv-BfVoq4g",
                "vQuality": "max"
            }).encode(),
            headers={
                "Accept": "application/json",
                "Content-Type": "application/json"
            }
        )
        res = urllib.request.urlopen(req)
        print("Cobalt instance (co.wuk.sh) response:", res.status)
        data = json.loads(res.read().decode())
        print(json.dumps(data, indent=2))
        return
    except Exception as e:
        print("Cobalt instance (co.wuk.sh) failed:", e)

    try:
        req = urllib.request.Request(
            'https://api.cobalt.tools/api/json',
            data=json.dumps({
                "url": "https://www.youtube.com/watch?v=2Vv-BfVoq4g",
                "vQuality": "max"
            }).encode(),
            headers={
                "Accept": "application/json",
                "Content-Type": "application/json"
            }
        )
        res = urllib.request.urlopen(req)
        print("Cobalt official response:", res.status)
        data = json.loads(res.read().decode())
        print(json.dumps(data, indent=2))
        return
    except Exception as e:
        print("Cobalt official failed:", e)

test_api()
